src/zino/events.py

```diff
@@ -147,20 +148,25 @@ def commit(self, event: Event, user: str = "monitor"):
 
         self._call_observers_for(new_event=event, old_event=old_event)
 
-    def delete(self, event: Event):
-        """Removes an event from the events dict and notifies all observers"""
-        try:
-            del self.events[event.id]
-        except KeyError:
-            pass
+    def _delete(self, event: Event):
+        """Removes a closed event from the events dict and notifies all observers"""
+        if event.state is EventState.CLOSED:
+            index = EventIndex(event.router, event.subindex, type(event))
+            if self._events_by_index.get(index) and event.id == self._events_by_index[index].id:
+                _log.info("Closed event %s was still in event index, removing it now", event.id)
+                del self._events_by_index[index]
+            try:
+                del self.events[event.id]
+            except KeyError:
+                pass
         self._call_observers_for(new_event=event)
 
     def delete_closed_events(self):
-        """Deletes all events that have been closed for at least 8 hours"""
+        """Deletes all events that have been closed for a certain time"""
```

Comment:
Actually, come to think of it, a better name for this is `delete_expired_events`.  This doesn't delete closed events, per se, it only deletes closed events that have *exipred*.